crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -45,9 +45,14 @@ pub trait Command: Send + Sync + 'static {
     fn write(self, world: &mut World);
 }
 
-/// A queue of [commands](Command) that get executed at the end of the stage of the system that called them.
+/// A [`Command`] queue to perform impactful changes to the [`World`].
 ///
-/// Commands are executed one at a time in an exclusive fashion.
+/// Since each command requires exclusive access to the `World`,
+/// all queued commands are automatically applied in sequence
+/// only after each system in a [stage] has completed.
+///
```

Comment:
The wording could probably be better and that sentence could be somewhere else, but I think we need to explicitly say somewhere that calling a function on Commands will only add a Command to the queue. It's implied by everything else, but anything that is implied is something that beginners might not understand.

```suggestion
///
/// Each function is only going to add a Command to the queue, it won't be instantly applied.
///
```

Reply:
> I think we need to explicitly say somewhere that calling a function on Commands will only add a Command to the queue.

The best place to do that is the methods' descriptions themselves. But I want to split that to a different PR, since I noted that it is best to keep docs PRs very tight-scoped.

Reply:
The methods should say that yes, but I think it's important to also say it in the struct doc because there's clearly some confusion about this. It feels like the current doc is pretty much already saying that, just not explicitly.

Reply:
I still think that it is redundant since the PR already states

> all queued commands are automatically applied in sequence only after each system in a stage has completed.

which is clear enough IMO. If people have wrong expectations about the behavior of the methods, it's because the docs for the methods are wrong.